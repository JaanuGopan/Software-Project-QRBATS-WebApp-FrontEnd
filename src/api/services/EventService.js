import axios from "axios";
import ApiConstants from "../config/ApiConstants";

class EventService {
  static async fetchEvents() {
    try {
      const response = await axios.post(ApiConstants.getAllEventUrl);
      return response.data;
    } catch (error) {
      console.error("Error fetching events:", error);
      throw error; // Re-throw the error to handle it in the component
    }
  }

  static async saveEvent(
    eventName,
    eventDate,
    eventValidDate,
    eventTime,
    eventEndTime,
    eventVenue,
    eventRole,
    moduleName,
    userId
  ) {
    try {
      const response = await axios.post(ApiConstants.createEventUrl, {
        eventName: eventName,
        eventDate: eventDate,
        eventValidDate: eventValidDate,
        eventTime: eventTime,
        eventEndTime: eventEndTime,
        eventVenue: eventVenue,
        eventRole: eventRole,
        eventModuleName: moduleName,
        eventAssignedUserId: userId,
      });
      return response.data;
    } catch (error) {
      console.error("Event creation failed", error);
    }
  }
}

export default EventService;
