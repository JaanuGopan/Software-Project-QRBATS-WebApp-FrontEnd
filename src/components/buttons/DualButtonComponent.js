import React, { useState } from "react";

const DualButtonComponent = ({ onSelect, buttonText1, buttonText2 }) => {
  const [selectedButton, setSelectedButton] = useState(1);

  const handleSelect = (button) => {
    setSelectedButton(button);
    onSelect(button);
  };

  return (
    <div className="row align-items-center">
      <div className="col align-self-center">
        <label>{`Select one`}</label>
      </div>
      <div className="col">
        <button
          className={`btn  ${
            selectedButton === 1 ? "btn-primary" : "btn-outline-primary"
          }`}
          onClick={() => handleSelect(1)}
        >
          {buttonText1}
        </button>
      </div>
      <div className="col">
        <button
          className={`btn ${
            selectedButton === 2 ? "btn-primary" : "btn-outline-primary"
          }`}
          onClick={() => handleSelect(2)}
        >
          {buttonText2}
        </button>
      </div>
    </div>
  );
};

export default DualButtonComponent;
