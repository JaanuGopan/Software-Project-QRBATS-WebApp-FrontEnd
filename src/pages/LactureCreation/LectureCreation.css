.lecture-creation-main-container{
    display: flex;
    flex-direction: column;
    overflow: scroll;
    margin-top: 15px;
    margin-left: 15px;
    margin-right: 5px;
    padding: 10px;
    width: 98%;
    height: 95%;
    border-top: 5px solid #0365A1;
    box-shadow: 0.5px 2px 0.5px 0.5px rgba(25, 65, 85, 0.5);
    border-radius: 10px;
    background-color: #FAFAFA;
}


.lecture-creation-title-icon-close-button{
    display: flex;
    flex-direction: row;
    align-self: center;
    margin-top: 10px;
    justify-self: center;
    align-items: center;
}

.lecture-creation-icon-close-button{
    color: red;
        cursor: pointer;
        display: flex;
        flex-direction: row;
        flex-wrap: nowrap;
        align-items: center;
        justify-content: center;
        padding: 5px;
        border-radius: 50px;
        margin-left: 97%;
        width: 35px;
        height: 35px;
}

.lecture-creation-icon-close-button:hover {
    color: white;
    background-color: red;

}



.lecture-creation-container{
    height: 100%;
    width: 100%;
    background-color: #FAFAFA;
    padding: auto;
    display: flex;
    gap: 10px;
}

.lecture-creation-container-left{
    height: 100%;
    width: 100%;
    background-color: #FAFAFA;
    padding: auto;  
}
.lecture-creation-container-right{
    height: 100%;
    width: 100%;
    background-color: #FAFAFA;
    padding: auto;
}
.left-container-lecture-creation{
    width: 100%;
    height: auto;
    display: flex;
    flex-direction: column;
    padding-left: 10%;
    padding-right: 10%;
    padding-top: 10px;
    padding-bottom: 20px;
    
}

.left-container-lecture-creation #selectModule, #toggle-button {
    margin-bottom: 15px;
}

.right-container-lecture-creation-heading{
    justify-self: center;
    align-self: center;
    display: flex;
    flex-direction: column;
    padding-top: 10px;
    padding-bottom: 10px;
}

.right-container-lecture-creation-heading label{
    justify-self: center;
    align-self: center;
}

.right-container-lecture-creation-day-time{
    width: 100%;
    height: 100%; 
}

.right-container-lecture-creation-toggle-button{
    align-items: center;
    justify-items: center;
    display: flex;
    flex-direction: column;
    margin: 10px;
    
}

.right-container-lecture-creation-time-selection-container {
    display: flex;
    flex-direction: column;
    border: 1px solid green;
    box-shadow: 0px 2px 5px 0px rgba(25, 65, 85, 0.5);
    border-radius: 5px;
    width: auto;
    margin-left: 20%;
    margin-right: 20%;
    overflow: scroll;
    max-height: 500px;
}

.right-container-lecture-creation-time-selection-time-slot{
    border: 0.7px solid #E65100;
    box-shadow: 0px 1.5px 4px 0px rgba(25, 65, 85, 0.5);
    width: 80%;
    border-radius: 5px;
    margin: auto;
    margin-top: 10px;
    margin-bottom: 10px;
}
.right-container-lecture-creation-time-selection-time-slot-heading{
    display: flex;
    flex-direction: row;
   align-items: center;
   justify-content: center;
   width: 100%;
}
.right-container-lecture-creation-time-selection-time-slot-heading #heading {
    color: #ED6C02;
    margin-left: 10px;
    width: auto;
}
.right-container-lecture-creation-time-selection-time-slot-heading #iconButton{
    align-self: last;
    margin-left: 50% ;
}



.right-container-lecture-creation-time-selection{
    width: auto;
    display: flex;
    flex-direction: row;
    padding: 10px;
    align-items: center;
    justify-content: center;
    justify-self: center;
    gap: 10px;
}

.right-container-lecture-creation-venue-selection-select{
    width: 50%;
}

.right-container-lecture-creation-time-selection-time-slot-button{
    width: 50%;
        display: flex;
        flex-direction: row;
        padding: 10px;
        align-items: center;
        justify-content: center;
        justify-self: center;
        margin: auto;
        margin-bottom: 10px;
}


.right-container-lecture-creation-submit-button{
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    margin-top: 20px;
    margin-bottom: 10px;
    width: 30%;
    align-self: center;
    justify-self: center;
    margin: 10px auto;
}
.left-container-lecture-creation-submit-button{
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    margin-top: 20px;
    margin-bottom: 10px;
    width: auto;
    align-self: center;
    justify-self: center;
    margin: 10px auto;
}

.lecture-creation-qrCode-popup{
    position: fixed;
        z-index: 4;
        left: 0;
        top: 0;
        overflow-y: scroll;
        width: 100%;
        height: 100vh;
        padding-top: 5vh;
        padding-bottom: 5vh;
        display: flex;
        align-items: center;
        justify-content: center;
        background-color: rgba(0, 0, 0, 0.5);
}

.lecture-qrCode-window{
    margin-top: 10px;
        width: 50%;
        border-top: 6px solid #0077C0;
        border-radius: 10px;
        padding: 20px;
        background-color: #ffffff;
        box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1);
        z-index: 4;
}

.lecture-qrCode-title-close-button {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    width: 100%;
}

.lecture-qrCode-title {
    align-self: center;
    justify-self: center;
    margin-left: auto;
}

.lecture-qrCode-close-button {
    align-self: center;
    justify-self: flex-end;
    margin-left: auto;
    height: 30px;
    width: 30px;
    align-items: center;
    justify-content: center;
    margin-bottom: 30px;
}

.lecture-qrCode-close-button #close-icon {
    align-self: center;
    justify-self: center;
    color: red;
    height: 90%;
    width: 90%;
}

.lecture-qrCode-close-button #close-icon:hover {
    align-self: center;
    justify-self: center;
    color: white;
    height: 90%;
    width: 90%;
    background-color: red;
    border-radius: 100%;
}

.lecture-qrCode-lecture-details-container{
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-evenly;
    padding: 10px;
}

.lecture-qrCode-container{
    display: flex;
    flex-direction: column;
    padding: auto;
    align-items: center;
    justify-content: center;

}

.lecture-qrCode{
    padding: 10;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 100%;
}

.lecture-qrCode-lectures-details{
    height: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

.lecture-qrCode-lectures-details-content{
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    padding: 1px;
}

.lecture-qrCode-lectures-details-content{
    display: flex;
    flex-direction: row;
    align-items: start;
    justify-content:start;
}

.available-lectures-container{
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: start;
    width: 100%;
    height: 45%;
}

.available-lecture-list{
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    width: 95%;
    height: 100%;
    /* border-top: 2px solid #0365A1;
    box-shadow: 0.5px 2px 0.5px 0.5px rgba(25, 65, 85, 0.5);
    border-radius: 10px; */
    background-color: #FAFAFA;
    margin-top: 10px;
    overflow: scroll;

}
.available-lecture-list-table{
    display: flex;
    flex-direction: column;
    justify-content: start;
    align-items: center;
    width: 95%;
    height: 100%;
    overflow: scroll;
    
}

.available-lecture-list-table-arrangement{
    box-shadow: 0px 5px 5px #ccc;
    width: 100%;
    max-width: 100%;
    height: auto;
    border-bottom: 2px solid #0176bf;
    overflow: scroll;
    margin-top: 10px;
    overflow: scroll;
    border-radius: 10px;

}

.available-lecture-list-table-arrangement thead{
    position: sticky;
    top: 0;
    background-color: #FFC107;
    color: #001f3f;
    border-bottom: 3px solid #c79604;
    text-align: center;
    border-radius: 10px;
}

.available-lecture-list-table-arrangement tr{
    border: 2px solid #FFC107;
    border-radius: 10px;
    max-height: 10vh;
}

.available-lecture-list-table-arrangement tbody{
    overflow: scroll;
}

.available-lecture-list-table-arrangement td{
    overflow: hidden;
    text-overflow: ellipsis;
    text-align: center;
    padding: 0.5rem;
    
}

.available-lecture-list-table-arrangement th{
    padding: 0.8rem;
    text-align: center;
}

.available-lecture-list-empty{
    align-self: center;
    justify-self: center;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 100%;
}




