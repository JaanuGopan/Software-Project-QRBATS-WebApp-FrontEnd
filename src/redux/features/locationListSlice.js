import { createSlice } from "@reduxjs/toolkit";

export const locationListSlice = createSlice({
  name: "locationList",
  initialState: ["NCC", "LT1", "LT2", "Auditorium", "DEIE", "DMME", "DCEE"],
  reducers: {
    setLocationList: (state, action) => {
      state = [];
      state.push(...state, action.payload);
    },
    resetLocationList: (state) => {
      state = [];
    },
  },
});

export const { setLocationList, resetLocationList } = locationListSlice.actions;
export default locationListSlice.reducer;
