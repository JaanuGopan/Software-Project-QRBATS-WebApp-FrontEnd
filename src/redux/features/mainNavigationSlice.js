import { createSlice } from "@reduxjs/toolkit";

export const mainNavigationSlice = createSlice({
  name: "mainNavigation",
  initialState: {
    sideBarIndex:
      localStorage.getItem("sideBarIndex") != null
        ? parseInt(localStorage.getItem("sideBarIndex"))
        : 0,
  },
  reducers: {
    setSideBarIndex: (state, action) => {
      state.sideBarIndex = action.payload;
      localStorage.setItem("sideBarIndex", action.payload);
    },
    resetSideBarIndex: (state) => {
      state.mainNavigation = 0;
      localStorage.removeItem("sideBarIndex");
    },
  },
});

export const { setSideBarIndex, resetSideBarIndex } =
  mainNavigationSlice.actions;
export default mainNavigationSlice.reducer;
