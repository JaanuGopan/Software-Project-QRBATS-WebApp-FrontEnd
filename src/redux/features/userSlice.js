import { createSlice } from '@reduxjs/toolkit';

export const userSlice = createSlice({
  name: 'user',
  initialState: {
    user: JSON.parse(localStorage.getItem('user')) || null,
    isLoggedIn: !!localStorage.getItem('authToken'),
    authToken: localStorage.getItem('authToken') || null,
  },
  reducers: {
    login: (state, action) => {
      state.user = action.payload;
    },
    logout: (state) => {
      state.user = null;
      localStorage.removeItem('user');
      localStorage.removeItem('authToken');
    },
  },
});

export const { login, logout } = userSlice.actions;
export const selectUser = (state) => state.user.user;
export default userSlice.reducer;
export const selectIsLoggedIn = (state) => state.user.isLoggedIn;
