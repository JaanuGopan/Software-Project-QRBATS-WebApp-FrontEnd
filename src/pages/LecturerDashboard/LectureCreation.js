import React, { useState, useEffect, useRef } from "react";
import EventService from "../../api/services/EventService";
import QRCode from "qrcode.react";
import { IoMdCloseCircleOutline } from "react-icons/io";
import toast, { Toaster } from "react-hot-toast";
import "../Event/EventCreation/EventCreation.css";
import eventCreationImage from "../../assets/Images/designer_pic/Designer_pic4.jpeg";
import { useSelector } from "react-redux";
import { selectUser } from "../../redux/features/userSlice";
import DualButtonComponent from "../../components/buttons/DualButtonComponent";
import Select from "react-select";
import ModuleService from "../../api/services/ModuleService";
import LocationService from "../../api/services/LocationService";

const LectureCreation = ({
  handleCloseCreateLectureWindow,
  reloadLectureList,
  hideCloseButton,
  locationNameList,
  showImage = false,
}) => {
  const user = useSelector(selectUser);
  const { userId, departmentId } = user || {};

  const [eventId, setEventId] = useState("");
  const [eventName, setEventName] = useState("");
  const [moduleName, setModuleName] = useState("");
  const [eventDate, setEventDate] = useState("");
  const [eventValidDate, setEventValidDate] = useState("");
  const [eventTime, setEventTime] = useState("");
  const [eventEndTime, setEventEndTime] = useState("");
  const [eventVenue, setEventVenue] = useState("");
  const [showQRCode, setShowQRCode] = useState(false);
  const [eventRole, setEventRole] = useState("LECTURE");
  const [eventAssignedUserId, setEventAssignedUserId] = useState(userId);
  const [title, setTitle] = useState("Lecture");
  const [moduleNameList, setModuleNameList] = useState([]);

  const [venueList, setVenuesList] = useState([
    "NCC",
    "LT1",
    "LT2",
    "Auditorium",
    "DEIE",
    "DMME",
    "DCEE",
  ]);

  const handleGetModulesList = async () => {
    const response = await ModuleService.getAllModulesByDepartmentId(
      departmentId
    );
    if (response) {
      const moduleList = response.map((module) => ({
        value: module.moduleId,
        label: module.moduleCode,
      }));
      setModuleNameList(moduleList);
      console.log(moduleList);
    }
  };

  const handleGetLocationNameList = async () => {
    try {
      const response = await LocationService.getAllLocationNames();
      setVenuesList(response);
    } catch (error) {
      console.error("Error fetching location names:", error);
    }
  };

  const locationList = [
    "NCC",
    "LT1",
    "LT2",
    "Auditorium",
    "DEIE",
    "DMME",
    "DCEE",
  ];

  const qrCodeRef = useRef(null);

  const notifySuccess = () => toast.success("Successfully Lecture Created!");

  const clearEventDetails = () => {
    setEventName("");
    setEventDate("");
    setEventValidDate("");
    setEventTime("");
    setEventEndTime("");
    setEventVenue("");
    setEventAssignedUserId(null);
    setEventRole("LECTURE");
    setModuleName(null);
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const response = await EventService.saveEvent(
        eventName,
        eventDate,
        eventValidDate,
        eventTime,
        eventEndTime,
        eventVenue,
        eventRole,
        eventRole === "LECTURE" ? moduleName.label : "",
        userId
      );
      setEventId(response.eventId);
      notifySuccess();
      reloadLectureList();
      clearEventDetails();
    } catch (error) {
      console.error("Lecture creation failed", error);
    } finally {
      setShowQRCode(true);
    }
  };

  const downloadQRCode = () => {
    const qrCodeURL = document
      .getElementById("qrCodeEl")
      .toDataURL("image/png")
      .replace("image/png", "image/octet-stream");
    let aEl = document.createElement("a");
    aEl.href = qrCodeURL;
    aEl.download = "QR_Code.png";
    document.body.appendChild(aEl);
    aEl.click();
    document.body.removeChild(aEl);
  };

  const eventDetails = {
    eventId: eventId,
    eventName: eventName,
    moduleName: moduleName,
    eventDate: eventDate,
    eventValidDate: eventValidDate,
    eventTime: eventTime,
    eventEndTime: eventEndTime,
    eventVenue: eventVenue,
    eventAssignedUserId: userId,
  };

  const qrCodeDetails = JSON.stringify(eventDetails);

  const [qrCodeWindow, setQrCodeWindow] = useState(false);

  const [selectedButton, setSelectedButton] = useState(1);

  const handleSelectedButton = (button) => {
    setSelectedButton(button);
    if (button === 2) {
      setEventRole("EVENT");
      setModuleName(null);
      setTitle("Event");
    } else {
      setEventRole("LECTURE");
      setTitle("Lecture");
    }
  };

  const today = new Date().toISOString().split("T")[0];

  const handleCloseQrCodeWindow = () => {
    setShowQRCode(false);
    clearEventDetails();
  };

  useEffect(() => {
    handleGetModulesList();
    handleGetLocationNameList();
    //setUserId(UserDetails.getUserId());
  }, []);

  return (
    <div className="event-main-container1">
      <Toaster />
      {!hideCloseButton && (
        <div
          className="closeCreateEventWindow"
          onClick={handleCloseCreateLectureWindow}
        >
          <IoMdCloseCircleOutline size={25} />
        </div>
      )}
      <h2>Create {` ${title}`}</h2>
      <div className="eventCreation-field">
        {showImage && (
          <img src={eventCreationImage} className="Create-logo" alt="Logo" />
        )}
        <div className="eventCreation-input-field">
          <DualButtonComponent
            onSelect={handleSelectedButton}
            buttonText1={"Lecture"}
            buttonText2={"Event"}
          />
          <form onSubmit={handleSubmit}>
            <div className="input-with-icon">
              <label className="date-label" htmlFor="moduleCode">
                {`${title} Name`}
              </label>
              <input
                required
                type="text"
                id="eventName"
                name="eventName"
                placeholder={`${title} Name`}
                className="form-control mb-2"
                value={eventName}
                onChange={(e) => setEventName(e.target.value)}
              />
            </div>

            {selectedButton === 1 && (
              <div className="input-with-icon">
                <label className="date-label" htmlFor="moduleCode">
                  {`Module Code`}
                </label>
                <Select
                  required
                  id="moduleName"
                  name="moduleName"
                  options={moduleNameList}
                  onChange={(e) => setModuleName(e)}
                  value={moduleName}
                  placeholder={"Select Module Code"}
                />
              </div>
            )}
            <div className="date-div">
              <div className="eventCreation-form">
                <div>
                  <label className="date-label" htmlFor="eventDate">
                    {`${title} Starting Date`}
                  </label>
                  <input
                    required
                    type="date"
                    min={today}
                    value={eventDate}
                    onChange={(e) => {
                      setEventDate(e.target.value);
                      setEventValidDate(e.target.value);
                    }}
                    className="form-control mb-2"
                  />
                </div>
              </div>
              {/* <div className="eventCreation-form">
                <div>
                  <label className="date-label" htmlFor="eventDate">
                    {`${title} Ending Date`}
                  </label>
                  <input
                    required
                    type="date"
                    min={today}
                    value={eventValidDate}
                    onChange={(e) => setEventValidDate(e.target.value)}
                    className="form-control mb-2"
                  />
                </div>
              </div> */}
            </div>
            <div className="eventCreation-form">
              <div>
                <label className="date-label" htmlFor="eventDate">
                  Venue
                </label>
                <select
                  required
                  value={eventVenue}
                  onChange={(e) => setEventVenue(e.target.value)}
                  className="form-control mb-2"
                >
                  <option value="">Select Venue</option>
                  {venueList.map((option, index) => (
                    <option key={index} value={option.locationName}>
                      {option.locationName}
                    </option>
                  ))}
                </select>
              </div>
            </div>

            <div className="eventCreation-form">
              <div>
                <label className="form-label" htmlFor="eventDate">
                  {`${title} Starting Time`}
                </label>
                <input
                  required
                  type="time"
                  value={eventTime}
                  onChange={(e) => setEventTime(e.target.value)}
                  placeholder={`${title} Time`}
                  className="form-control mb-2"
                />
              </div>
              <div className="eventCreation-form">
                <div>
                  <label className="form-label" htmlFor="eventEndTime">
                    {`${title} End Time`}
                  </label>
                  <input
                    required
                    type="time"
                    value={eventEndTime}
                    onChange={(e) => setEventEndTime(e.target.value)}
                    placeholder={`${title} End Time`}
                    className="form-control mb-2"
                  />
                </div>
              </div>
            </div>

            <div className="eventCreation-form"></div>
            <button
              onClick={() => setQrCodeWindow(true)}
              type="submit"
              className="btn btn-success w-100"
            >
              Create {` ${title}`}
            </button>
          </form>
        </div>
        {qrCodeWindow && showQRCode && (
          <div className="Admin-Create-Event-Dashboard">
            <div ref={qrCodeRef} className="event-main-container1">
              <div
                className="closeCreateEventWindow"
                onClick={handleCloseQrCodeWindow}
              >
                <IoMdCloseCircleOutline />
              </div>
              <h2>Successfully Event Created</h2>
              <div className="row-center">
                <QRCode
                  name="QRCode"
                  size={200}
                  id="qrCodeEl"
                  value={qrCodeDetails}
                  className="mb-2"
                />
              </div>
              <div>
                <p className="text-center">Scan this QR code to join {title}</p>
              </div>
              <div className="row-center">
                <div className="QRbutton">
                  <button
                    onClick={downloadQRCode}
                    className="btn btn-success mr-3"
                  >
                    Save
                  </button>
                </div>
              </div>
            </div>
          </div>
        )}
      </div>
    </div>
  );
};

export default LectureCreation;
