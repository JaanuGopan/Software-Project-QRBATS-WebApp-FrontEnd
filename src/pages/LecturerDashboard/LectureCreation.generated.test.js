import renderer from 'react-test-renderer';
import React, { useState, useEffect, useRef } from "react";
import EventService from "../../api/services/EventService";
import QRCode from "qrcode.react";
import { IoMdCloseCircleOutline } from "react-icons/io";
import toast, { Toaster } from "react-hot-toast";

import eventCreationImage from "../../assets/Images/signin/Signin.jpeg";
import html2canvas from "html2canvas";
import jsPDF from "jspdf";
import { useSelector } from "react-redux";
import { selectUser } from "../../redux/features/userSlice";
import DualButtonComponent from "../../components/buttons/DualButtonComponent";
import LectureCreation from "./LectureCreation";

jest.mock("../../api/services/EventService");
jest.mock("qrcode.react");
jest.mock("react-icons/io");
jest.mock("react-hot-toast");
jest.mock("../Event/EventCreation/EventCreation.css");
jest.mock("../../assets/Images/signin/Signin.jpeg");
jest.mock("html2canvas");
jest.mock("jspdf");
jest.mock("react-redux");
jest.mock("../../redux/features/userSlice");
jest.mock("../../components/buttons/DualButtonComponent");

const renderTree = tree => renderer.create(tree);
describe('<LectureCreation>', () => {
  it('should render component', () => {
    expect(renderTree(<LectureCreation 
    />).toJSON()).toMatchSnapshot();
  });
  
});