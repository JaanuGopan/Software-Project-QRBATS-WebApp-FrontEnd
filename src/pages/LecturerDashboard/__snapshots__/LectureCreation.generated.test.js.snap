// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LectureCreation> should render component 1`] = `
<div
  className="event-main-container1"
>
  <div
    className="closeCreateEventWindow"
  />
  <h2>
    Create 
     Lecture
  </h2>
  <div
    className="eventCreation-field"
  >
    <img
      alt="Logo"
      className="Create-logo"
      src="Signin.jpeg"
    />
    <div
      className="eventCreation-input-field"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="input-with-icon"
        >
          <input
            className="form-control mb-2"
            id="eventName"
            name="eventName"
            onChange={[Function]}
            placeholder="Lecture Name"
            required={true}
            type="text"
            value=""
          />
        </div>
        <div
          className="input-with-icon"
        >
          <input
            className="form-control mb-2"
            id="moduleName"
            name="moduleName"
            onChange={[Function]}
            placeholder="Module Name"
            required={true}
            type="text"
            value={null}
          />
        </div>
        <div
          className="date-div"
        >
          <div
            className="eventCreation-form"
          >
            <div>
              <label
                className="date-label"
                htmlFor="eventDate"
              >
                Lecture Starting Date
              </label>
              <input
                className="form-control mb-2"
                onChange={[Function]}
                required={true}
                type="date"
                value=""
              />
            </div>
          </div>
          <div
            className="eventCreation-form"
          >
            <div>
              <label
                className="date-label"
                htmlFor="eventDate"
              >
                Lecture Ending Date
              </label>
              <input
                className="form-control mb-2"
                onChange={[Function]}
                required={true}
                type="date"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          className="eventCreation-form"
        >
          <div>
            <label
              className="date-label"
              htmlFor="eventDate"
            >
              Venue
            </label>
            <select
              className="form-control mb-2"
              onChange={[Function]}
              required={true}
              value=""
            >
              <option
                value=""
              >
                Select Venue
              </option>
              <option
                value="NCC"
              >
                NCC
              </option>
              <option
                value="LT1"
              >
                LT1
              </option>
              <option
                value="LT2"
              >
                LT2
              </option>
              <option
                value="Auditorium"
              >
                Auditorium
              </option>
              <option
                value="DEIE"
              >
                DEIE
              </option>
              <option
                value="DMME"
              >
                DMME
              </option>
              <option
                value="DCEE"
              >
                DCEE
              </option>
              <option
                value="Other"
              >
                Other
              </option>
            </select>
          </div>
        </div>
        <div
          className="eventCreation-form"
        >
          <div>
            <label
              className="form-label"
              htmlFor="eventDate"
            >
              Lecture Starting Time
            </label>
            <input
              className="form-control mb-2"
              onChange={[Function]}
              placeholder="Lecture Time"
              required={true}
              type="time"
              value=""
            />
          </div>
          <div
            className="eventCreation-form"
          >
            <div>
              <label
                className="form-label"
                htmlFor="eventEndTime"
              >
                Lecture End Time
              </label>
              <input
                className="form-control mb-2"
                onChange={[Function]}
                placeholder="Lecture End Time"
                required={true}
                type="time"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          className="eventCreation-form"
        />
        <button
          className="btn btn-success w-100"
          onClick={[Function]}
          type="submit"
        >
          Create 
           Lecture
        </button>
      </form>
    </div>
  </div>
</div>
`;
